doc/source/ray-contribute/development.rst

```diff
@@ -74,39 +133,55 @@ For RHELv8 (Redhat EL 8.0-64 Minimal), run the following commands:
   sudo yum groupinstall 'Development Tools'
   sudo yum install psmisc
 
-Install bazel manually from link: https://docs.bazel.build/versions/main/install-redhat.html 
-
+In RedHat, install bazel manually from this link: https://docs.bazel.build/versions/main/install-redhat.html
 
-For MacOS, run the following commands:
+Preparing to build Ray on MacOS
+-------------------------------
 
-.. tip:: Assuming you already have brew and bazel installed on your mac and you also have grpc and protobuf installed on your mac consider removing those (grpc and protobuf) for smooth build through commands ``brew uninstall grpc``, ``brew uninstall protobuf``. If you have built the source code earlier and it still fails with error as ``No such file or directory:``, try cleaning previous builds on your host by running commands ``brew uninstall binutils`` and ``bazel clean --expunge``.
+.. tip:: Assuming you already have brew and bazel installed on your mac and you also have grpc and protobuf installed on your mac consider removing those (grpc and protobuf) for smooth build through the commands ``brew uninstall grpc``, ``brew uninstall protobuf``. If you have built the source code earlier and it still fails with errors like ``No such file or directory:``, try cleaning previous builds on your host by running the commands ``brew uninstall binutils`` and ``bazel clean --expunge``.
 
+To build Ray on MacOS, first install these dependencies:
 
 .. code-block:: bash
 
   brew update
   brew install wget
 
-Ray can be built from the repository as follows.
+  # Install Bazel.
+  ray/ci/env/install-bazel.sh
 
-.. code-block:: bash
+Building Ray on Linux & MacOS (full)
+------------------------------------
 
-  git clone https://github.com/ray-project/ray.git
+Make sure you have a local clone of the git repository as explained above.
 
-  # Install Bazel.
-  ray/ci/env/install-bazel.sh
-  # (Windows users: please manually place Bazel in your PATH, and point
-  # BAZEL_SH to MSYS2's Bash: ``set BAZEL_SH=C:\Program Files\Git\bin\bash.exe``)
+You will also need to install NodeJS to build the dashboard: https://nodejs.org
+
+Enter into the project directory, for example:
+
+.. code-block:: shell
+
+    cd ray
+
+Now you can build the dashboard. From inside of your local Ray project directory run:
+
+.. code-block:: bash
 
-  # Build the dashboard
-  # (requires Node.js, see https://nodejs.org/ for more information).
-  pushd ray/dashboard/client
+  pushd dashboard/client
   npm install
   npm run build
   popd
 
+Now let's build Ray for Python.
+
+Make sure you activate any Python virtual (or conda) environment you could be using as described above.
+
+Enter into the directory for Python inside of the Ray project directory and install the project with ``pip``:
```

Comment:
maybe say `python` here for clarity (specifying that this is indeed the folder name you mean)

Reply:
Good point! I used `python/` to make it explicit that I'm referring to the directory.